### ISSUE TITLE
Create Entry button is available for collection that is disabled on a multisite

### ISSUE THREAD

**Issue by stuartcusackie** [statamic/cms#3756]
## Bug Description

I have a collection that should only exists on one domain, and only one domain is chosen in the collection configuration under 'sites'.

The problem is that the collection shows on the all domains regardless of the settings and it leads to a broken 'Create Entry'  button that leads to a blank form. (screenshot):

Ideally this collection shouldn't be shown at all when viewing the other sites. Or alternatively the create entry button wouldn't be available on those sites.

## How to Reproduce
Set up a multisite and collection that is only available on one site.

## Extra Detail
![image](https://user-images.githubusercontent.com/20210934/119165849-db599600-ba55-11eb-822e-86b1f42b084b.png)

## Environment

**Statamic version:** 3.1.14

**PHP version:** 8.0

**Install method (choose one):**
- Existing Laravel app

**Comment by stuartcusackie** [comment]
I've just realised that I may be going against the flow of statamic here. This still seems like a bug but I can avoid it with a differenct configuration.

I want a collection that is identical on all sites, so that's why I removed the other sites in the collection's settings - I was then pulling in the collection from the origin site using a custom controller.

I have now update the collection to be available on all sites, which works better with Statamic's logic, and the bug I originally posted won't be seen. The only downside is now the user has to go through the publish process for all sites when creating these entries. It's not the end of the world but perhaps there could be an easier way to autopublish entries from specific collections on all sites?
